File: .circleci/env.py
#!/usr/bin/env python

import os

branch = os.environ.get('CIRCLE_BRANCH')

if branch == 'master':
  os.environ['NG_ENV'] = 'production'

elif branch == 'stage':
  os.environ['NG_ENV'] = 'staging'

elif branch == 'qa':
  os.environ['NG_ENV'] = 'qa'

else:
  os.environ['NG_ENV'] = 'development'

print("export NG_ENV=\"" + os.environ.get('NG_ENV') + "\"")

File: .circleci/env.js
#!/usr/bin/env node

switch (process.env.CIRCLE_BRANCH) {
  case 'master':
    process.env.NG_ENV = 'production';
    break;

  case 'staging':
    process.env.NG_ENV = 'staging';
    break;

  case 'qa':
    process.env.NG_ENV = 'qa';
    break;

  default:
    process.env.NG_ENV = 'development';
    break;
}

console.log(`export NG_ENV="${process.env.NG_ENV}"`);
